ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -111,35 +120,37 @@ public SchemaKStream into(
             }
           }
           return new GenericRow(columns);
-        }).to(kafkaTopicName, Produced.with(Serdes.String(), topicValueSerDe));
+        }).to(kafkaTopicName, Produced.with(keySerde, topicValueSerDe));
     return this;
   }
 
   @SuppressWarnings("unchecked")
-  public SchemaKStream filter(final Expression filterExpression) {
-    final SqlPredicate predicate = new SqlPredicate(filterExpression, schema, false,
+  public SchemaKStream<K> filter(final Expression filterExpression) {
+    final SqlPredicate predicate = new SqlPredicate(filterExpression, schema, isWindowed(),
```

Comment:
`KStream` is not windowed and `isWindowed` should always be `false`.

Reply:
see above.